src/Features/CSharp/Portable/Completion/CompletionProviders/CrefCompletionProvider.cs

```diff
@@ -35,178 +43,267 @@ public override bool IsTriggerCharacter(SourceText text, int characterPosition,
+            //   <see cref="M[x, out |
+
+            if (!token.Parent.IsKind(SyntaxKind.CrefParameterList, SyntaxKind.CrefBracketedParameterList))
+            {
+                return false;
+            }
+
+            if (token.IsKind(SyntaxKind.OpenParenToken) &&
+                token.Parent.IsKind(SyntaxKind.CrefParameterList))
+            {
+                return true;
             }
 
-            return await CreateItemsAsync(document.Project.Solution.Workspace, semanticModel,
-                position, result, token, cancellationToken).ConfigureAwait(false);
+            if (token.IsKind(SyntaxKind.OpenBracketToken) &&
+                token.Parent.IsKind(SyntaxKind.CrefBracketedParameterList))
+            {
+                return true;
+            }
+
+            return token.IsKind(SyntaxKind.CommaToken, SyntaxKind.RefKeyword, SyntaxKind.OutKeyword);
         }
 
-        private bool IsSignatureContext(SyntaxToken token)
+        private static bool IsCrefQualifiedNameContext(SyntaxToken token)
         {
-            return token.IsKind(SyntaxKind.OpenParenToken, SyntaxKind.CommaToken, SyntaxKind.RefKeyword, SyntaxKind.OutKeyword)
-                && token.Parent.IsKind(SyntaxKind.CrefParameterList, SyntaxKind.CrefBracketedParameterList);
+            // cases:
+            //   <see cref="x.|
+
+            return token.IsKind(SyntaxKind.DotToken)
+                && token.Parent.IsKind(SyntaxKind.QualifiedCref);
         }
 
-        // LookupSymbols doesn't return indexers or operators because they can't be referred to by name, so we'll have to try to 
-        // find the innermost type declaration and return its operators and indexers
-        private IEnumerable<ISymbol> GetOperatorsAndIndexers(SyntaxToken token, SemanticModel semanticModel, CancellationToken cancellationToken)
+        private static IEnumerable<ISymbol> GetSymbols(SyntaxToken token, SemanticModel semanticModel, CancellationToken cancellationToken)
         {
-            var typeDeclaration = token.GetAncestor<TypeDeclarationSyntax>();
+            if (IsCrefStartContext(token))
+            {
+                return GetUnqualifiedSymbols(token, semanticModel, cancellationToken);
+            }
+            else if (IsCrefParameterListContext(token))
+            {
+                return semanticModel.LookupNamespacesAndTypes(token.SpanStart);
+            }
+            else if (IsCrefQualifiedNameContext(token))
+            {
+                return GetQualifiedSymbols((QualifiedCrefSyntax)token.Parent, token, semanticModel, cancellationToken);
+            }
+
+            return SpecializedCollections.EmptyEnumerable<ISymbol>();
+        }
 
-            var result = new List<ISymbol>();
+        private static IEnumerable<ISymbol> GetUnqualifiedSymbols(SyntaxToken token, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            foreach (var symbol in semanticModel.LookupSymbols(token.SpanStart))
```

Comment:
I'm not sure I follow. A foreach and yield is hardly complex and would box the ImmutableArray returned by LookupSymbols.